mobile/native/src/trade/position/handler.rs

```diff
@@ -51,47 +53,90 @@ pub fn set_position_state(state: PositionState) -> Result<()> {
     Ok(())
 }
 
-/// A channel renewal could be triggered to:
-///
-/// - Roll over (no offer associated).
-/// - Open a new position.
-/// - Resize a position.
-pub fn handle_channel_renewal_offer(expiry_timestamp: OffsetDateTime) -> Result<()> {
+pub fn handle_renew_offer() -> Result<()> {
     if let Some(position) = db::get_positions()?.first() {
-        // Assume that if there is an order in filling we are dealing with position resizing.
-        //
-        // TODO: This has caused problems in the past. Any other ideas? We could generate
-        // `ProtocolId`s using `OrderId`s whenever possible on the coordinator, and compare the two
-        // values here to be sure.
-        if db::get_order_in_filling()?.is_some() {
-            tracing::debug!("Setting position to resizing");
-
-            let position =
-                db::update_position_state(position.contract_symbol, PositionState::Resizing)?;
-
-            event::publish(&EventInternal::PositionUpdateNotification(position));
-        }
-        // Without an order, we must be rolling over.
-        else {
-            tracing::debug!("Setting position to rollover");
-
-            db::rollover_position(position.contract_symbol, expiry_timestamp)?;
+        tracing::debug!("Received renew offer to resize position");
 
-            let mut position = position.clone();
-            position.position_state = PositionState::Rollover;
-            position.expiry = expiry_timestamp;
+        let position =
+            db::update_position_state(position.contract_symbol, PositionState::Resizing)?;
 
-            event::publish(&EventInternal::PositionUpdateNotification(position));
-        }
+        event::publish(&EventInternal::PositionUpdateNotification(position));
     } else {
         // If we have no position, we must be opening a new one.
-        tracing::info!("Received channel renewal proposal to open new position");
+        tracing::info!("Received renew offer to open new position");
+    }
+
+    Ok(())
+}
+
+pub fn handle_rollover_offer(
+    expiry_timestamp: OffsetDateTime,
+    funding_fee_events: &[FundingFeeEvent],
+) -> Result<()> {
+    tracing::debug!("Setting position state to rollover");
+
+    let positions = &db::get_positions()?;
+    let position = positions.first().context("No position to roll over")?;
+
+    // TODO: Do we actually need to update the expiry early? Ideally, we would set it after the
```

Comment:
Agreed

Reply:
Do you think it's safe for me to change this then? I guess nothing crucial depends on this.

Reply:
I think it's safe to change.